src/args.rs

```diff
@@ -0,0 +1,157 @@
+            .default_value(""))
+    .arg(
+        Arg::from_usage(
+            "--slowdown-gpio=[0..4] 'Slowdown GPIO. Needed for faster Pis/slower panels'")
+            .default_value("1"))
+
+    // Flags
+    .arg(
+        Arg::from_usage(
+            "--show-refresh 'Show refresh rate'"))
+    .arg(
+        Arg::from_usage(
+            "--inverse 'Switch if your matrix has inverse colors on'"))
+    .arg(
+        Arg::from_usage(
+            "--no-hardware-pulse 'Don't use hardware pin-pulse generation'"))
+    .arg(
+        Arg::from_usage(
+            "--daemon 'Make the process run in the background as daemon'"))
+    .arg(
+        Arg::from_usage(
+            "--no-drop-privs 'Don't drop privileges from 'root' after initializing the hardware'"))
+}
+
+pub fn matrix_options_from_args<'a>(parsed_args: &clap::ArgMatches<'a>) -> LedMatrixOptions {
+    let mut options = LedMatrixOptions::new();
+
+    let gpio_mapping = parsed_args.value_of("gpio-mapping").unwrap();
+    let rows = value_t!(parsed_args, "rows", u32).unwrap();
+    let cols = value_t!(parsed_args, "cols", u32).unwrap();
+    let chain = value_t!(parsed_args, "chain", u32).unwrap();
+    let parallel = value_t!(parsed_args, "parallel", u32).unwrap();
+    let multiplexing = value_t!(parsed_args, "multiplexing", u32).unwrap();
+    let pixel_mapper = parsed_args.value_of("pixel-mapper").unwrap();
+    let pwm_bits = value_t!(parsed_args, "pwm-bits", u8).unwrap();
+    let brightness = value_t!(parsed_args, "brightness", u8).unwrap();
+    let scan_mode = value_t!(parsed_args, "scan-mode", u32).unwrap();
+    let row_addr_type = value_t!(parsed_args, "row-addr-type", u32).unwrap();
+    let limit_refresh = value_t!(parsed_args, "limit-refresh", u32).unwrap();
+    let rgb_sequence = parsed_args.value_of("rgb-sequence").unwrap();
+    let pwm_lsb_nanoseconds = value_t!(parsed_args, "pwm-lsb-nanoseconds", u32).unwrap();
+    let pwm_dither_bits = value_t!(parsed_args, "pwm-dither-bits", u32).unwrap();
+    let panel_type = parsed_args.value_of("panel-type").unwrap();
+    let _slowdown_gpio = value_t!(parsed_args, "slowdown-gpio", u32).unwrap();
+
+    // flags
+    let show_refresh: bool = parsed_args.is_present("show-refresh");
+    let inverse: bool = parsed_args.is_present("inverse");
+    let no_hardware_pulse: bool = parsed_args.is_present("no-hardware-pulse");
+    let _daemon: bool = parsed_args.is_present("daemon");
+    let _no_drop_privs: bool = parsed_args.is_present("no-drop-privs");
+
+    let scan_mode_bool: bool = match scan_mode {
+        0 => false,
+        _ => true,
+    };
+    let parallel_bool: bool = match parallel {
+        0 => false,
+        _ => true,
+    };
+
+    options.set_hardware_mapping(gpio_mapping);
+    options.set_rows(rows);
+    options.set_cols(cols);
+    options.set_chain_length(chain);
+    options.set_parallel(parallel_bool);
```

Comment:
I think the parallel API in general is wrong. Perhaps it was a boolean back when this project started, but you can now have up to three now. Probably should fix that now...